src/pages/home/sidebar/SidebarLinks.js

```diff
@@ -49,47 +45,43 @@ const propTypes = {
 
     isLoading: PropTypes.bool.isRequired,
 
+    // eslint-disable-next-line react/require-default-props
     priorityMode: PropTypes.oneOf(_.values(CONST.PRIORITY_MODE)),
 
     isActiveReport: PropTypes.func.isRequired,
-
-    ...withLocalizePropTypes,
-};
-
-const defaultProps = {
-    priorityMode: CONST.PRIORITY_MODE.DEFAULT,
 };
 
-class SidebarLinks extends React.PureComponent {
-    constructor(props) {
-        super(props);
+function SidebarLinks({onLinkClick, insets, optionListItems, isLoading, priorityMode = CONST.PRIORITY_MODE.DEFAULT, isActiveReport, isCreateMenuOpen}) {
+    const modal = useRef({});
+    const {translate} = useLocalize();
+    const {isSmallScreenWidth} = useWindowDimensions();
 
-        this.showSearchPage = this.showSearchPage.bind(this);
-        this.showReportPage = this.showReportPage.bind(this);
-
-        if (this.props.isSmallScreenWidth) {
-            App.confirmReadyToOpenApp();
+    useEffect(() => {
+        if (!isSmallScreenWidth) {
+            return;
         }
-    }
+        App.confirmReadyToOpenApp();
+    }, [isSmallScreenWidth]);
 
-    componentDidMount() {
+    useEffect(() => {
         App.setSidebarLoaded();
         SidebarUtils.setIsSidebarLoadedReady();
-        this.isSidebarLoaded = true;
 
-        let modal = {};
-        this.unsubscribeOnyxModal = onyxSubscribe({
+        const unsubscribeOnyxModal = onyxSubscribe({
             key: ONYXKEYS.MODAL,
             callback: (modalArg) => {
-                modal = modalArg;
+                if (_.isNull(modalArg)) {
+                    return;
+                }
```

Comment:
To prevent 
```js
if (modal.current.willAlertModalBecomeVisible) {
         return;
}
```
from erroring out when the modal is null in onyx initially (Safeguarding the modal ref)